<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<title>org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch():.36fa30bc7272755055a4879635c9b3e9_node7</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<style type="text/css">
body { background-color:#fff; color:#000; font-family:Helvetica, sans-serif; font-size:10pt }
h1 { font-size:14pt }
.code { border-collapse:collapse; width:100%; font-family: "Andale Mono", monospace; font-size:10pt; line-height: 1.2em }
.color_black { color: black }
.color_blue { color: blue }
.color_green { color: green }
.color_orange { color: orange }
.color_red { color: red }
.line { border-left: 3px solid #ccc; padding-left: 1ex; white-space: pre; }
.msg { background-color:#fff8b4; color:#000; float:left; font-family:Helvetica, sans-serif; font-size: smaller; font-weight: bold; margin-bottom:10px; margin-top:10px; max-width:60em; padding:0.5em 1ex 0.5em 1ex; -webkit-border-radius:5px; -webkit-box-shadow:1px 1px 7px #000; white-space: pre-wrap; word-wrap: break-word; }
.num { background-color:#eee; color:#444; font-size: smaller; padding-right:2ex; text-align:right; width:2.5em; }
.dangling { background-color:white; color: gray; }
.visited { background-color:LemonChiffon; color: darkmagenta; }
.tooltip { display: none; background-color:#FFF0F0; border: 2px solid #F00; font-weight: normal; left:10em; padding: 2px; position: absolute; top: -1em; -webkit-border-radius:5px; -webkit-box-shadow:1px 1px 7px #000; z-index: 1}
.with_tooltip { position: relative; }
.with_tooltip:hover .tooltip, .visited:hover .tooltip { display: block; }
</style>
</head>
<body>
<center><h1>Cfg Node <a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">7</a></h1></center>PROC: <a href="./../org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9.html">void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch()</a> LINE: <a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>
<br>PREDS:
<a href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node15.html"><span class='visited with_tooltip'>N_15<span class='tooltip'>node15 preds:17 succs:7 exn:3 </span></span></a><br>SUCCS:
<a href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node8.html"><span class='visited with_tooltip'>N_8<span class='tooltip'>node8 preds:7 succs:9 exn:3 </span></span></a><br>EXN:
<a href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node3.html"><span class='visited with_tooltip'>N_3<span class='tooltip'>node3 preds:16 succs:2 exn:2 </span></span></a><br>
<LISTING class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915];
*&amp;error:java.lang.Object*=n$6 [line 915];
</LISTING>
<hr width="100%">
<a name="session14" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session14">node7#session14</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) [compute pre] liveness 7
<LISTING class='color_black'>JOIN

LEFT:   normal:{ error, npe }, exn:{ }
RIGHT:  normal:{ }, exn:{ }
RESULT: normal:{ error, npe }, exn:{ }

</LISTING>
<hr width="100%">
<a name="session15" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session15">node7#session15</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) [exec] liveness 7
<LISTING class='color_black'>PRE STATE:
normal:{ error, npe }, exn:{ }

INSTR=  *&amp;error:java.lang.Object*=n$6 [line 915]; 

STATE:
normal:{ n$6, npe }, exn:{ }

INSTR=  n$6=*&amp;CatchVar22:java.lang.Object* [line 915]; 

STATE:
normal:{ CatchVar22, npe }, exn:{ }

</LISTING>
<hr width="100%">
<a name="session49" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session49">node7#session49</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) [compute pre] nullify
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session50" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session50">node7#session50</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) [exec] nullify
<LISTING class='color_black'>PRE STATE:
(normal:{ CatchVar22, npe }, exn:{ },normal:{ n$22, n$23 }, exn:{ })

INSTR=  n$6=*&amp;CatchVar22:java.lang.Object* [line 915]; 

STATE:
(normal:{ n$6, CatchVar22, npe }, exn:{ },normal:{ n$22, n$23 }, exn:{ })

INSTR=  *&amp;error:java.lang.Object*=n$6 [line 915]; 

STATE:
(normal:{ error, npe }, exn:{ },normal:{ n$6, CatchVar22 }, exn:{ })

</LISTING>
<hr width="100%">
<a name="session126" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session126">node7#session126</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session127" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session127">node7#session127</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session128" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session128">node7#session128</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session129" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session129">node7#session129</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session34" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session34">node7#session34</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) biabduction
<LISTING class='color_black'>**** [FP:Pending] void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch() Node: 7, Procedure: void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_red'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
  [footprint
     <span class='color_red'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
  .... Instructions: ....
  <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915];
  *&amp;error:java.lang.Object*=n$6 [line 915];
  NULLIFY(&amp;CatchVar22); [line 915];
  EXIT_SCOPE(n$6,CatchVar22); [line 915];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915]</span>
    in vpath_find exp:n$149
    vpath_find: cannot find n$149
    in vpath_find exp:n$144
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$143
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$143
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$149); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$150) ;
    error = n$145<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$146<span class='color_orange'>initial</span>:void; CatchVar22 = n$143<span class='color_orange'>update:915</span>:void; $irvar4 = n$147<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$148<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$144<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError
    [footprint
       this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )  ]</span>
    
    in prop_iter_extend_ptsto lexp: &amp;CatchVar22
    prop before:
    <span class='color_blue'>CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$149); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$150) ;
    error = n$145<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$146<span class='color_orange'>initial</span>:void; CatchVar22 = n$143<span class='color_orange'>update:915</span>:void; $irvar4 = n$147<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$148<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$144<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError
    [footprint
       this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )  ]</span>
    
    
    prop list after:
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>CatchVar22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$149)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (n$150)</span> ;
    <span class='color_blue'>error = n$145<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = n$146<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = n$143<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar4 = n$147<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = n$148<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = n$144<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
    
    
    
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:void</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>*&amp;error:java.lang.Object*=n$6 [line 915]</span>
    in vpath_find exp:n$156
    vpath_find: cannot find n$156
    in vpath_find exp:n$151
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$6
    .... Rearrangement Start ....
    Exp: &amp;error
    Prop:
    <span class='color_blue'>CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$156); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$157) ;
    error = n$152<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$153<span class='color_orange'>initial</span>:void; CatchVar22 = n$6<span class='color_orange'>update:915</span>:void; $irvar4 = n$154<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$155<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$151<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError
    [footprint
       this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )  ]</span>
    
    in prop_iter_extend_ptsto lexp: &amp;error
    prop before:
    <span class='color_blue'>CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$156); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$157) ;
    error = n$152<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$153<span class='color_orange'>initial</span>:void; CatchVar22 = n$6<span class='color_orange'>update:915</span>:void; $irvar4 = n$154<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$155<span class='color_orange'>initial</span>:void; this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$151<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError
    [footprint
       this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )  ]</span>
    
    
    prop list after:
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>CatchVar22 != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$156)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (n$157)</span> ;
    <span class='color_blue'>error = n$152<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = n$153<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = n$6<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar4 = n$154<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = n$155<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = n$151<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
    
    
    
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = n$6<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>NULLIFY(&amp;CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>error != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = n$6<span class='color_orange'>update:915</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$6,CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>error != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>error != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>
    [footprint
       <span class='color_blue'>this = @f$0<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>  ]</span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session50" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session50">node7#session50</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch() Node: 7, Procedure: void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_red'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
  .... Instructions: ....
  <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915];
  *&amp;error:java.lang.Object*=n$6 [line 915];
  NULLIFY(&amp;CatchVar22); [line 915];
  EXIT_SCOPE(n$6,CatchVar22); [line 915];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915]</span>
    in vpath_find exp:n$149
    vpath_find: cannot find n$149
    in vpath_find exp:n$144
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$143
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$143
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$149); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$150) ;
    error = n$145<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$146<span class='color_orange'>initial</span>:void; CatchVar22 = n$143<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$147<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$148<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$144<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;error:java.lang.Object*=n$6 [line 915]</span>
    in vpath_find exp:n$156
    vpath_find: cannot find n$156
    in vpath_find exp:n$151
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$6
    .... Rearrangement Start ....
    Exp: &amp;error
    Prop:
    <span class='color_blue'>CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$156); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$157) ;
    error = n$152<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$153<span class='color_orange'>initial</span>:void; CatchVar22 = n$6<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$154<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$155<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$151<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = n$6<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>NULLIFY(&amp;CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>error != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = n$6<span class='color_orange'>update:915</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$6,CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>error != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>error != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session64" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session64">node7#session64</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch() Node: 7, Procedure: void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_red'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span></span>
  .... Instructions: ....
  <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915];
  *&amp;error:java.lang.Object*=n$6 [line 915];
  NULLIFY(&amp;CatchVar22); [line 915];
  EXIT_SCOPE(n$6,CatchVar22); [line 915];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915]</span>
    in vpath_find exp:n$376
    vpath_find: cannot find n$376
    in vpath_find exp:n$371
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$370
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$370
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0; CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$376); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$377) ;
    error = n$372<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$373<span class='color_orange'>initial</span>:void; CatchVar22 = n$370<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$374<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$375<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$371<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof); npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;error:java.lang.Object*=n$6 [line 915]</span>
    in vpath_find exp:n$383
    vpath_find: cannot find n$383
    in vpath_find exp:n$378
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$6
    .... Rearrangement Start ....
    Exp: &amp;error
    Prop:
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0; CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$383); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$384) ;
    error = n$379<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$380<span class='color_orange'>initial</span>:void; CatchVar22 = n$6<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$381<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$382<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$378<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof); npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = n$6<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>NULLIFY(&amp;CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>error != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = n$6<span class='color_orange'>update:915</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_blue'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$6,CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>error != 0</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>error != 0</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$3<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session78" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session78">node7#session78</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch() Node: 7, Procedure: void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_red'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span></span>
  .... Instructions: ....
  <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915];
  *&amp;error:java.lang.Object*=n$6 [line 915];
  NULLIFY(&amp;CatchVar22); [line 915];
  EXIT_SCOPE(n$6,CatchVar22); [line 915];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915]</span>
    in vpath_find exp:n$603
    vpath_find: cannot find n$603
    in vpath_find exp:n$598
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$597
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$597
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0; CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$603); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$604) ;
    error = n$599<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$600<span class='color_orange'>initial</span>:void; CatchVar22 = n$597<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$601<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$602<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$598<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof); npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;error:java.lang.Object*=n$6 [line 915]</span>
    in vpath_find exp:n$610
    vpath_find: cannot find n$610
    in vpath_find exp:n$605
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$6
    .... Rearrangement Start ....
    Exp: &amp;error
    Prop:
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0; CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$610); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$611) ;
    error = n$606<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$607<span class='color_orange'>initial</span>:void; CatchVar22 = n$6<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$608<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$609<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$605<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof); npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = n$6<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>NULLIFY(&amp;CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>error != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = n$6<span class='color_orange'>update:915</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span>; <span class='color_blue'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$6,CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>error != 0</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>error != 0</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar|->{}<span class='color_orange'>formal</span>:java.lang.String( sub )(instof)</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session92" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session92">node7#session92</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch() Node: 7, Procedure: void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_red'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
  .... Instructions: ....
  <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915];
  *&amp;error:java.lang.Object*=n$6 [line 915];
  NULLIFY(&amp;CatchVar22); [line 915];
  EXIT_SCOPE(n$6,CatchVar22); [line 915];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915]</span>
    in vpath_find exp:n$832
    vpath_find: cannot find n$832
    in vpath_find exp:n$827
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$826
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$826
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0; CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$832); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$833) ;
    error = n$828<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$829<span class='color_orange'>initial</span>:void; CatchVar22 = n$826<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$830<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$831<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$827<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;error:java.lang.Object*=n$6 [line 915]</span>
    in vpath_find exp:n$839
    vpath_find: cannot find n$839
    in vpath_find exp:n$834
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$6
    .... Rearrangement Start ....
    Exp: &amp;error
    Prop:
    <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0; CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$839); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$840) ;
    error = n$835<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$836<span class='color_orange'>initial</span>:void; CatchVar22 = n$6<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$837<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$838<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$834<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = n$6<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>NULLIFY(&amp;CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>error != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = n$6<span class='color_orange'>update:915</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$6,CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>error != 0</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>error != 0</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session106" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session106">node7#session106</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch() Node: 7, Procedure: void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_red'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
  .... Instructions: ....
  <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915];
  *&amp;error:java.lang.Object*=n$6 [line 915];
  NULLIFY(&amp;CatchVar22); [line 915];
  EXIT_SCOPE(n$6,CatchVar22); [line 915];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915]</span>
    in vpath_find exp:n$1061
    vpath_find: cannot find n$1061
    in vpath_find exp:n$1056
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$1055
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$1055
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$1061); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$1062) ;
    error = n$1057<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$1058<span class='color_orange'>initial</span>:void; CatchVar22 = n$1055<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$1059<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$1060<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$1056<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;error:java.lang.Object*=n$6 [line 915]</span>
    in vpath_find exp:n$1068
    vpath_find: cannot find n$1068
    in vpath_find exp:n$1063
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$6
    .... Rearrangement Start ....
    Exp: &amp;error
    Prop:
    <span class='color_blue'>CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$1068); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$1069) ;
    error = n$1064<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$1065<span class='color_orange'>initial</span>:void; CatchVar22 = n$6<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$1066<span class='color_orange'>initial</span>:void; $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$1067<span class='color_orange'>initial</span>:void; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$1063<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = n$6<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>NULLIFY(&amp;CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>error != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = n$6<span class='color_orange'>update:915</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$6,CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>error != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>error != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session120" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session120">node7#session120</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch() Node: 7, Procedure: void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_red'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
  .... Instructions: ....
  <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915];
  *&amp;error:java.lang.Object*=n$6 [line 915];
  NULLIFY(&amp;CatchVar22); [line 915];
  EXIT_SCOPE(n$6,CatchVar22); [line 915];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915]</span>
    in vpath_find exp:n$1290
    vpath_find: cannot find n$1290
    in vpath_find exp:n$1285
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$1284
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$1284
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$1290); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$1291) ;
    error = n$1286<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$1287<span class='color_orange'>initial</span>:void; CatchVar22 = n$1284<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$1288<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$1289<span class='color_orange'>initial</span>:void; this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$1285<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;error:java.lang.Object*=n$6 [line 915]</span>
    in vpath_find exp:n$1297
    vpath_find: cannot find n$1297
    in vpath_find exp:n$1292
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$6
    .... Rearrangement Start ....
    Exp: &amp;error
    Prop:
    <span class='color_blue'>CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$1297); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$1298) ;
    error = n$1293<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$1294<span class='color_orange'>initial</span>:void; CatchVar22 = n$6<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$1295<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$1296<span class='color_orange'>initial</span>:void; this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$1292<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = n$6<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>NULLIFY(&amp;CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>error != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = n$6<span class='color_orange'>update:915</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$6,CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>error != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>error != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = 0<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$5<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$4<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session134" href="./../nodes/org.junit.tests.assertion.AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch%28%29%3A.36fa30bc7272755055a4879635c9b3e9_node7.html#session134">node7#session134</a>(<a href="./../../AssertionTest.java.8ae87590c7f481b71a4d70b019da36d9.html#LINE915">915</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch() Node: 7, Procedure: void AssertionTest.assertThrowsSuppliesACoherentErrorMessageUponTypeMismatch(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$6)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_red'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$7)</span> ;
  <span class='color_red'>error = _t$5<span class='color_orange'>initial</span>:java.lang.AssertionError*</span>; <span class='color_red'>$irvar3 = _t$4<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>CatchVar22 = _t$3<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_red'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_red'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_red'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_red'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_red'>this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_red'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
  <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_red'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
  .... Instructions: ....
  <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915];
  *&amp;error:java.lang.Object*=n$6 [line 915];
  NULLIFY(&amp;CatchVar22); [line 915];
  EXIT_SCOPE(n$6,CatchVar22); [line 915];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$6=*&amp;CatchVar22:java.lang.Object* [line 915]</span>
    in vpath_find exp:n$1517
    vpath_find: cannot find n$1517
    in vpath_find exp:n$1512
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$1511
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$1511
    .... Rearrangement Start ....
    Exp: &amp;CatchVar22
    Prop:
    <span class='color_blue'>CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$1517); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$1518) ;
    error = n$1513<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$1514<span class='color_orange'>initial</span>:void; CatchVar22 = n$1511<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$1515<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$1516<span class='color_orange'>initial</span>:void; this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$1512<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>initial</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_red'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>*&amp;error:java.lang.Object*=n$6 [line 915]</span>
    in vpath_find exp:n$1524
    vpath_find: cannot find n$1524
    in vpath_find exp:n$1519
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find do_sexp: no match on Eexp &amp;CatchVar22
    vpath_find: cannot find n$6
    .... Rearrangement Start ....
    Exp: &amp;error
    Prop:
    <span class='color_blue'>CatchVar22 != 0; MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(n$1524); MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe); MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22); RET &lt; NullPointerException.<init>() &gt; (n$1525) ;
    error = n$1520<span class='color_orange'>initial</span>:java.lang.AssertionError*; $irvar3 = n$1521<span class='color_orange'>initial</span>:void; CatchVar22 = n$6<span class='color_orange'>update:915</span>:void; $RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub ); $irvar4 = n$1522<span class='color_orange'>initial</span>:void; $RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub ); $irvar0 = 0<span class='color_orange'>nullify</span>:void; return = null<span class='color_orange'>update:915</span>:void; $irvar1 = 0<span class='color_orange'>nullify</span>:void; $irvar2 = n$1523<span class='color_orange'>initial</span>:void; this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub ); npe = n$1519<span class='color_orange'>update:911</span>:java.lang.NullPointerException* ;
    npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException; CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>CatchVar22 != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(CatchVar22)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = <span class='color_red'>n$6<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = n$6<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>CatchVar22|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>NULLIFY(&amp;CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>error != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$4)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$5)</span> ;
    <span class='color_blue'>error = n$6<span class='color_orange'>update:915</span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span>; <span class='color_blue'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$6,CatchVar22); [line 915]</span>
    in vpath_find exp:_t$4
    vpath_find: cannot find _t$4
    in vpath_find exp:_t$0
    vpath_find: found npe :java.lang.NullPointerException*
    in vpath_find exp:n$6
    vpath_find: found error :java.lang.AssertionError*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>error != 0</span>; <span class='color_blue'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_blue'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_red'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = _t$3<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>CatchVar22 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_red'>error != 0</span>; <span class='color_red'>MEMne &lt; ThrowingRunnable AssertionTest.throwingRunnable(NullPointerException):914 &gt; <span class='color_orange'>{vpath: }</span>(_t$5)</span>; <span class='color_red'>MEMne &lt; Throwable Assert.assertThrows(Class,AssertionTest$3):914 &gt; <span class='color_orange'>{vpath: }</span>(error)</span>; <span class='color_blue'>MEMne &lt; new:911 &gt; <span class='color_orange'>{vpath: }</span>(npe)</span>; <span class='color_blue'>RET &lt; NullPointerException.<init>() &gt; (_t$6)</span> ;
    <span class='color_blue'>error = <span class='color_red'>_t$4<span class='color_orange'>update:915</span></span>:java.lang.AssertionError*</span>; <span class='color_blue'>$irvar3 = <span class='color_red'>_t$3<span class='color_orange'>initial</span></span>:void</span>; <span class='color_blue'>CatchVar22 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getMessage():java.lang.String|abducedRetvar = val$8<span class='color_orange'>formal</span>:java.lang.String*( sub )</span>; <span class='color_blue'>$irvar4 = _t$2<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>$RET_java.lang.AssertionError.getCause():java.lang.Throwable|abducedRetvar = val$7<span class='color_orange'>formal</span>:java.lang.Throwable*( sub )</span>; <span class='color_blue'>$irvar0 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:915</span>:void</span>; <span class='color_blue'>$irvar1 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$irvar2 = _t$1<span class='color_orange'>initial</span>:void</span>; <span class='color_blue'>this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>old_this = val$6<span class='color_orange'>formal</span>:org.junit.tests.assertion.AssertionTest*( sub )</span>; <span class='color_blue'>npe = _t$0<span class='color_orange'>update:911</span>:java.lang.NullPointerException*</span> ;
    <span class='color_red'>error|->{}<span class='color_orange'>alloc</span>:java.lang.AssertionError</span>; <span class='color_blue'>npe|->{}<span class='color_orange'>alloc</span>:java.lang.NullPointerException</span></span>
    
    
    
  </LISTING>